Add `--compression` flag to `influx write` (#2243)
influxdata/docs-v2: content/influxdb/v2.0/reference/cli/influx/write/_index.md

| [dryrun](/influxdb/v2.0/reference/cli/influx/write/dryrun) | Write to stdout instead of InfluxDB |

## Flags
| Flag |                     | Description                                                                                  | Input type  | {{< cli/mapped >}}    |
|:-----|:--------------------|:---------------------------------------------------------------------------------------------|:-----------:|:----------------------|
| `-c` | `--active-config`   | CLI configuration to use for command                                                         | string      |                       |
| `-b` | `--bucket`          | Bucket name (mutually exclusive with `--bucket-id`)                                          | string      | `INFLUX_BUCKET_NAME`  |
|      | `--bucket-id`       | Bucket ID (mutually exclusive with `--bucket`)                                               | string      | `INFLUX_BUCKET_ID`    |
|      | `--configs-path`    | Path to `influx` CLI configurations (default `~/.influxdbv2/configs`)                        | string      | `INFLUX_CONFIGS_PATH` |

> |      | `--compression`     | Input compression, either `none` or `gzip`. Defaults to `none` unless input ends with `.gz`. | string      |                       |

Author: This is the new line. ⬆️

> |      | `--compression`     | Input compression, either `none` or `gzip`. Defaults to `none` unless input ends with `.gz`. | string      |                       |

Reviewer: ```suggestion
|      | `--compression`     | Input compression (`none` or `gzip`, default is `none` unless input file ends with `.gz`. | string      |                       |
```

| `-p` | `--precision`       | Precision of the timestamps (default `ns`)                                                   | string      | `INFLUX_PRECISION`    |
|      | `--rate-limit`      | Throttle write rate (examples: `5 MB / 5 min` or `1MB/s`).                                   | string      |                       |
|      | `--skip-verify`     | Skip TLS certificate verification                                                            |             |                       |
|      | `--skipHeader`      | Skip first *n* rows of input data                                                            | integer     |                       |
|      | `--skipRowOnError`  | Output CSV errors to stderr, but continue processing                                         |             |                       |
| `-t` | `--token`           | Authentication token                                                                         | string      | `INFLUX_TOKEN`        |
| `-u` | `--url`             | URL to import data from                                                                      | stringArray |                       |

## Examples

> 

Reviewer: We need to add examples for both line protocol and CSV. Add the following examples in their respective sections as well as links in the Examples TOC.

~~~md
##### Write line protocol from a compressed file
```sh
# The influx CLI assumes files with the .gz extension use gzip compression 
influx write \
  --bucket example-bucket \
  --file path/to/line-protocol.txt.gz

# Specify gzip compression for gzipped files without the .gz extension
influx write \
  --bucket example-bucket \
  --file path/to/line-protocol.txt.comp \
  --compression gzip
```

##### Write annotated CSV data from a compressed file
```sh
# The influx CLI assumes files with the .gz extension use gzip compression 
influx write \
  --bucket example-bucket \
  --file path/to/data.csv.gz

# Specify gzip compression for gzipped files without the .gz extension
influx write \
  --bucket example-bucket \
  --file path/to/data.csv.comp \
  --compression gzip
```
~~~
